script.js

```diff
@@ -0,0 +1,61 @@
+let bookCollection = {};
```

Comment:
The need to get the keys to the collection indicates that the data structure that would hold the books makes more sense as an array of objects:

```js
[
  { 
    author: "ABC",
    book: "bookabc"
  },
  { 
    author: "ABCD",
    book: "bookabcd"
  }
]
```